File: LinkedLists/2. Add Two Numbers.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def addTwoNumbers(self, l1, l2):
        number1 = ""
        number2 = ""

        while l1 != None:
            number1 += (str(l1.val))
            l1 = l1.next
    
        while l2 != None:
            number2 += (str(l2.val))
            l2 = l2.next

        number1 = number1[::-1]
        number2 = number2[::-1]  
        
        return int(number1) + int(number2)
    


class Solution:
    def addTwoNumbers(self, l1, l2):
        head = None
        pointer = None
        sum = 0

        while l1 != None and l2 != None:
            sum += l1.val + l2.val
            if head == None:
                head = ListNode(sum%10)
                pointer = head

            else:
                node = ListNode(sum%10)
                pointer.next = node
                pointer = pointer.next

            sum //= 10

            l1 = l1.next
            l2 = l2.next
        

        while l1 != None:
            sum += l1.val
            node = ListNode(sum%10)
            pointer.next = node
            pointer = pointer.next

            sum //= 10
            l1 = l1.next

        while l2 != None:
            sum += l2.val
            node = ListNode(sum%10)
            pointer.next = node
            pointer = pointer.next

            sum //= 10
            l2 = l2.next

        if(sum != 0):
            pointer.next = ListNode(sum)

        return head

File: LinkedLists/2.Add Two Numbers.cpp
// /**
//  * Definition for singly-linked list.
//  * struct ListNode {
//  *     int val;
//  *     ListNode *next;
//  *     ListNode() : val(0), next(nullptr) {}
//  *     ListNode(int x) : val(x), next(nullptr) {}
//  *     ListNode(int x, ListNode *next) : val(x), next(next) {}
//  * };
//  */


struct ListNode 
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {};
    ListNode(int x) : val(x), next(nullptr) {};
    ListNode(int x, ListNode *next) : val(x), next(next){};
};


class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode* head = new ListNode(0);
        ListNode* current = head;
        int sum = 0;
        while (l1 != nullptr && l2 != nullptr)
        {
            int x = l1 ? l1->val : 0;
            int y = l2 ? l2->val : 0;
            sum += x + y;
            current->next = new ListNode(sum % 10);
            sum = sum  / 10;
            
            current = current->next;
            l1 = l1 ? l1->next : nullptr;
            l2 = l2 ? l2->next : nullptr;
        }
        

        while(l1 != nullptr){
            sum += l1 ? l1->val : 0;
            current->next = new ListNode(sum%10);
            sum = sum / 10;
            l1 = l1 ? l1->next : nullptr;
            current = current->next;

        }


        while(l2 != nullptr){
            sum += l2 ? l2->val : 0;

            current->next = new ListNode(sum%10);
            sum = sum / 10;
            l2 = l2 ? l2->next : nullptr;
            current = current->next;
        }

        if(sum != 0){
            current->next = new ListNode(sum%10);
        }

        return head->next;

    }
};